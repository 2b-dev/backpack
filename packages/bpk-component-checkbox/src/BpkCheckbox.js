import React, { PropTypes } from 'react'

import './bpk-checkbox.scss'

const BpkCheckbox = ({ name, label, value, id, checked, onChange, disabled }) => {
  const classNames = [ 'bpk-checkbox' ]

  disabled ? classNames.push('bpk-checkbox--disabled') : null

  return (
    <label className={classNames.join(' ')}>
      <input
        type='checkbox'
        className='bpk-checkbox__input'
        name={name}
        value={value || name}
        id={id}
        checked={checked}
        onChange={onChange}
        disabled={disabled}
      />
      {label ? ` ${label}` : null}
    </label>
  )
}

BpkCheckbox.propTypes = {
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  id: PropTypes.string,
  checked: PropTypes.bool,
  value: PropTypes.string,
  label: PropTypes.string,
  disabled: PropTypes.bool
}

BpkCheckbox.defaultProps = {
  id: null,
  checked: false,
  value: null,
  label: null,
  disabled: false
}

export default BpkCheckbox
