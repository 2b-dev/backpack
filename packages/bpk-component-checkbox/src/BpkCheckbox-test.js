import React from 'react';
import renderer from 'react-test-renderer';
import BpkCheckbox from './BpkCheckbox';

describe('BpkCheckbox', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <BpkCheckbox
        name="checkbox"
        label="Prefer directs"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with id attribute', () => {
    const tree = renderer.create(
      <BpkCheckbox
        name="checkbox"
        label="Prefer directs"
        id="checkbox"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with checked attribute', () => {
    const tree = renderer.create(
      <BpkCheckbox
        name="checkbox"
        label="Prefer directs"
        checked
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with disabled attribute', () => {
    const tree = renderer.create(
      <BpkCheckbox
        name="checkbox"
        label="Prefer directs"
        disabled
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with white attribute', () => {
    const tree = renderer.create(
      <BpkCheckbox
        name="checkbox"
        label="Prefer directs"
        white
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with value attribute', () => {
    const tree = renderer.create(
      <BpkCheckbox
        name="checkbox"
        label="Prefer directs"
        value="my-value"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

