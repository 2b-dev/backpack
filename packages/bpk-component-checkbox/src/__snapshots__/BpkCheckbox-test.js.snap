exports[`BpkCheckbox should render correctly 1`] = `
<label
  className="bpk-checkbox">
  <input
    className="bpk-checkbox__input"
    disabled={false}
    name="checkbox"
    onChange={[Function]}
    type="checkbox"
    value="checkbox" />
</label>
`;

exports[`BpkCheckbox should render correctly with checked attribute 1`] = `
<label
  className="bpk-checkbox">
  <input
    checked={true}
    className="bpk-checkbox__input"
    disabled={false}
    name="checkbox"
    onChange={[Function]}
    type="checkbox"
    value="checkbox" />
</label>
`;

exports[`BpkCheckbox should render correctly with disabled attribute 1`] = `
<label
  className="bpk-checkbox bpk-checkbox--disabled">
  <input
    className="bpk-checkbox__input"
    disabled={true}
    name="checkbox"
    onChange={[Function]}
    type="checkbox"
    value="checkbox" />
</label>
`;

exports[`BpkCheckbox should render correctly with id attribute 1`] = `
<label
  className="bpk-checkbox">
  <input
    className="bpk-checkbox__input"
    disabled={false}
    id="checkbox"
    name="checkbox"
    onChange={[Function]}
    type="checkbox"
    value="checkbox" />
</label>
`;

exports[`BpkCheckbox should render correctly with label attribute 1`] = `
<label
  className="bpk-checkbox">
  <input
    className="bpk-checkbox__input"
    disabled={false}
    name="checkbox"
    onChange={[Function]}
    type="checkbox"
    value="checkbox" />
   My label
</label>
`;

exports[`BpkCheckbox should render correctly with value attribute 1`] = `
<label
  className="bpk-checkbox">
  <input
    className="bpk-checkbox__input"
    disabled={false}
    name="checkbox"
    onChange={[Function]}
    type="checkbox"
    value="my-value" />
</label>
`;
